index.js

```diff
@@ -19,14 +19,26 @@ module.exports = {
     'no-unused-vars': ['off'],
     // Use typescript version of "no-unused-vars" rule, because the default rule doesn't recognize certain typescript
     // features like enumerations
-    '@typescript-eslint/no-unused-vars': ['warn', { argsIgnorePattern: '^_', varsIgnorePattern: '^_' }],
-    'no-use-before-define': ['error', { functions: false }],
+    '@typescript-eslint/no-unused-vars': ['warn', { varsIgnorePattern: '^_', args: 'none' }],
+    'no-use-before-define': ['warn'],
+
+    // Prettier takes care of max length in code and eslint warns about prettier warnings due to the rule
+    // 'prettier/prettier'. Setting max-len to 120 for code here would add some linter warnings in code where prettier
+    // does not obey to the 120 char line length, so we're "supressing" it with 1000...
+    'max-len': ['warn', { code: 1000, comments: 120 }],
```

Comment:
TBH, I don't understand the second sentence.
I mean the length of 120 is already checked by prettier and forwarded to ESLint, so why add another rule here?

Reply:
If only using `'max-len': ['warn', { comments: 120 }],`, eslint falls back to max line length of 80 for code and warns about all lines > 80 chars despite the `prettier/prettier` rule and if using `'max-len': ['warn', { code: 120, comments: 120 }],` eslint still warns about some lines where prettier does not obey to the 120 char max like the following:

**app-header.tsx**
```tsx
      {loggedIn && (
        <div className="flex flex-row items-center gap-3">
          { /* ESLint would warn about line length of 156 here where prettier does not obey the 120 char max */ }
          <div className="w-[40px] h-[40px] bg-neutral-100 rounded-lg flex items-center justify-center text-[1.75rem] font-semibold uppercase select-none">
            {userInfo.firstname?.[0] || userInfo.email?.[0]}
          </div>
          <div className="flex flex-col">
            <div className="text-m-medium font-semibold">
              {userInfo.firstname} {userInfo.lastname}
            </div>
            <div className="text-s-medium">{userInfo.email}</div>
          </div>
        </div>
      )}
```

In such situations we could either "help" prettier stay <= 120 chars e.g. by extracting the classes into a dedicated variable etc. or just accept some exceptions from the 120 char max rule. I went for the 2nd option but I'm open to consider the alternative if you'd prefer this.

I'll at least try to adjust the comment to make it more clear for future us 😉.

Reply:
I see, thanks for the explanatation 👍